File: python/demos/demo_pure_pursuit.py
#!/usr/bin/env python3
"""Pure Pursuit demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_sst.py's precedent (the
# package re-export lives in navigation/local_planning/tracking/__init__.py).
from navigation.local_planning.tracking.pure_pursuit import PurePursuit

if __name__ == "__main__":
    run_local("pure_pursuit", PurePursuit)

File: cpp/demos/demo_pure_pursuit.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/tracking/pure_pursuit.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "pure_pursuit", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::PurePursuitPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
